### PR TITLE
Added ExceptionInterface as marker for package-specific exception

### PR DESCRIPTION
**Pull request by michalbundyra**
All exceptions thrown in the repository should implement that interface.
Added suffix "Trait" and "Interface" to exception traits and interfaces.

### CODE REVIEW

File: src/Exception/ProblemDetailsExceptionInterface.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
 /**
  * Defines an exception type for generating Problem Details.
  */
-interface ProblemDetailsException extends JsonSerializable
+interface ProblemDetailsExceptionInterface extends ExceptionInterface, JsonSerializable
```
Comment (weierophinney) [inline src/Exception/ProblemDetailsExceptionInterface.php]: This interface should not extend the package marker interface; it's intended for _consumers_ to implement, and, as such, will fall within their own package inheritance.

The `ExceptionInterface` marker is for identifying which exceptions _are thrown by this package_.

**Review by weierophinney — CHANGES_REQUESTED** [review]
I have one concern, and it's a big one: the `ProblemDetailsExceptionInterface` SHOULD NOT extend the package `ExceptionInterface`, which I detail more in a comment below.

**Comment by weierophinney** [comment]
@webimpress Don't forget to update the docs as well; I noticed it first in those. :smile:

**Comment by michalbundyra** [comment]
@weierophinney All updated right now 😄
Rebased to resolve conflicts.

**Review by weierophinney — APPROVED** [review]

**Comment by weierophinney** [comment]
Thanks, @webimpress

**Pull request merged**